### PR TITLE
Add help target to Makefile

### PR DESCRIPTION
**Pull request by stmcginnis**
**Issue number:**

N/A

**Description of changes:**

This adds a `help` target and sets it as the default. This helps newer
contributors discover the make targets of relevance by providing a
nicely formatted output of the targets they might be interested in.

This can be expanded as we go to add additional targets or to categorize
targets into different groups based on their expected usage.

Also some minor cleanup to the Makefile itself.

**Testing done:**

image

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
 TOOLS_IMAGE ?= public.ecr.aws/bottlerocket-test-system/bottlerocket-test-tools:$(BOTTLEROCKET_TOOLS_VERSION)
 
+# Capture information about the build host
+TESTSYS_BUILD_HOST_UNAME_ARCH=$(shell uname -m)
+TESTSYS_BUILD_HOST_GOARCH ?= $(lastword $(subst :, ,$(filter $(TESTSYS_BUILD_HOST_UNAME_ARCH):%,x86_64:amd64 aarch64:arm64)))
+TESTSYS_BUILD_HOST_PLATFORM=$(shell uname | tr '[:upper:]' '[:lower:]')
+# On some hosts we get an x509 certificate error and need to set GOPROXY to "direct"
+TESTSYS_BUILD_GOPROXY ?= direct
+
+# The set of images to create. Add new artifacts here when added to the
```
Comment (ecpullen) [inline Makefile]: Maybe note that these are for bottlerocket images and the example images have their own targets.

File: Makefile [hunk 2]
```diff
@@ -153,15 +165,15 @@ cargo-deny:
 	cargo deny --all-features --no-default-features check --disable-fetch licenses sources
 
 # Define a target to tag all images
-tag-images: $(TAG_IMAGES)
+tag-images: $(TAG_IMAGES)  ## tag all images
```
Comment (ecpullen) [inline Makefile:168]: Should we put the env variables for registry and tag?

**Review by ecpullen — APPROVED** [review]
LGTM

File: Makefile (hunk 2 above)
Comment (stmcginnis) [reply Makefile:168]: Not sure I understand, can you elaborate?

Comment (ecpullen) [reply Makefile:168]: The registry can be configured for tagging with `PUBLISH_IMAGES_REGISTRY` and the tag can be configured with `PUBLISH_IMAGES_VERSION`

Comment (stmcginnis) [reply Makefile:168]: That gets to be a bit long to add detailed help text. Do you mind if we start with the more concise help output and look at better ways of providing these details as a follow on?

**Comment by stmcginnis** [comment]
Rebased on `develop` to pick up the latest updates.

**Review by etungsten — APPROVED** [review]

**Pull request merged**